docs/en/plugins/README.md

```diff
@@ -0,0 +1,413 @@
+- You can inject any class you'd like into the plugin controllers, which can be defined either in the core repo or within your plugin.
+- `Records` is a collection of all of our records interfaces, and `find` a method we can use to search for records within a given interface.
+
+```
+# plugins/loomio_tags/components/thread_tags/thread_tags.haml
+.thread-tags{ng-if: "anyTags()"}
+  .thread-tag{ng-repeat: "tag in tags()", style: "background-color: {{tag.color}};"}
+    %span {{tag.name}}
+```
+
+_NB: notice that these need to be in a folder within a /components folder in your plugin, and that that folder must be named the same as each file._
+
+Then, we can tell our plugin about the component we created, and what outlet we want it to live in, using the `use_component` command
+
+```
+plugin.use_component :thread_tags, outlet: :before_thread_title
+```
+
+_NB: We support using the same outlet for multiple components, so don’t worry about conflicts there._
+
+You can also put a single component into multiple outlets in the interface by passing an array of names:_
+
+```
+plugin.use_component :thread_tags, outlet: [:before_thread_title, :after_thread_title]
+```
+
+##### 4. Fetching the tags from the server
+The only thing left is to make an API call to fetch our tags! In order to do this, we'll create another directive, which specifically performs an API request when it's loaded. (This is the recommended way to load remote records into your app.)
+
+I've called it `tag_fetcher`, and it has just a single file, `tag_fetcher.coffee`
+
+```
+# plugins/loomio_tags/components/tag_fetcher.coffee
+angular.module('loomioApp').directive 'tagFetcher', ->
+  restrict: 'E'
+  replace: true
+  controller: ($scope, Records, DiscussionTagRecordsInterface) ->
+    Records.addRecordsInterface(DiscussionTagRecordsInterface) if !Records.discussionTags
+    Records.discussionTags.fetch
+      params:
+        discussion_keys: _.pluck($scope.query.threads(), 'key')
+```
+
+_NB: Notice that we're adding the records interface to Records if it doesn't exist yet._
+
+Then, we pop this into another outlet in `thread_preview_collection.haml`:
+
+```
+%outlet{name: "before-thread-previews"}
+```
+```
+plugin.use_component :tag_fetcher, outlet: :before_thread_previews
+```
+
+Now, if we load up our app, we’ll see our tag displayed nicely next to the discussion title. Hooray!
+
+### PART 5: Testing our controller
+
+If you're not testing it, you're not doing it right! We provide the ability to add your own specs. If they're in the plugins/ folder, they'll even be run as part of the CI, so you can be more confident that your plugins aren't breaking stuff.
+
+For this plugin, the thing that makes most sense is a controller spec, to make sure that we're responding with the right thing when an API call is made. You can check it out [here](https://github.com/loomio/loomio_tags/blob/master/spec/discussion_tags_controller_spec.rb)
```

Comment:
For this plugin, the thing that makes most sense is a controller spec, to make sure that we're responding with the right thing when an API call is made. You can check it out [here](https://github.com/loomio/loomio_tags/blob/master/spec/discussion_tags_controller_spec.rb)**.** [add fullstop]